text/0000-orderer-snapshot.md

```diff
@@ -0,0 +1,263 @@
+OSN ledger snapshot functionality is designed with the same principles as the peer ledger snapshot, in particular it is a full snapshot. We do not implement an incremental snapshot capability. Each snapshot is fully independent and self-sufficient and does not rely on the content of any previous snapshot. For example even if we start an OSN from a snapshot and take a new snapshot and there is no new config block in between, we do not refer to the last config block from the previous snapshot, instead we include the copy of the same last config block in the new snapshot.
+
+## Guide: Create snapshot
+
+To create a ledger snapshot use the following REST API (block_number must be supplied as a parameter in the request body):
+
+    POST /v1/mgmt/channels/{channelID}/snapshot
+
+ or the following CLI command:
+
+    $ osnadmin snapshot submitrequest -c <channelID> -b <block_number>
+
+A process of snapshot creation will start immediately if the block number is less than current ledger height. If the block number is in future (>= ledger height) then the operation will immediately fail with a reasonable error message.
+
+The process of snapshot creation is an asynchronous one, so the success of the "snapshot submitrequest" operation does not indicate that the snapshot itself is ready. The indication of the snapshot readiness is the presence of the appropriate directory in the file system (<Config.FileLedger.snapshots.rootDir>/completed/<channelID>/<block_number>).
+
+## Guide: Join channel from a snapshot
+
+To join a channel from a snapshot use the following REST API (files from the snapshot must be placed in the body, using multipart/form-data content-type):
+
+    POST /v1/mgmt/channels
+
+ or the following CLI command:
+
+    $ osnadmin channel joinbysnapshot -s <snapshot_dir>
+
+To check the status of the "join by snapshot" operation use the following REST API:
+
+    GET /v1/mgmt/channels/{channelID}
+
+ or the following CLI command:
+
+    $ osnadmin channel list -c <channelID>
+
+## Guide: Prune block ledger of an existing channel
+[guide-prune-block-ledger-of-an-existing-channel]: #guide-prune-block-ledger-of-an-existing-channel
+
+To prune an existing channel ledger do the following:
+ - create a new snapshot at the desired block number,
+ - remove the OSN from the channel,
+ - join the channel from the snapshot.
+
+This approach has some drawbacks, see [Drawbacks section](#drawbacks).
+
+# Reference-level explanation
+[reference-level-explanation]: #reference-level-explanation
+
+## Changes to the existing API
+
+This RFC renames the existing **channel participation API** to **channel management API** and introduces three new logical operations (see [Guide-level explanation](#guide-level-explanation)) at this API while keeping the existing channel participation operations, namely:
```

Comment:
I'm not sure that it's appropriate to do this renaming.

The term "channel management API" was considered and debated during the development of the original feature, with the concern being that "channel management" conveys the idea that the API has an affect on the configuration of the channel, it does not.  This API only has an affect on how the orderer interacts with that channel.  Particularly, whether the orderer participates in the channel or not.

Extending this API to support not only the participation actions, but also the snapshotting/pruning actions might make this old name a bit misleading, but there's certainly a lot of breaking for existing applications and documentation by doing so.

Have you considered simply exposing another rest path specific to the new functions?

Reply:
Done, here:
507e634c0711d2668e93ae6aa5f8cc0ad2798973